firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistribution.java

```diff
@@ -455,4 +479,18 @@ private UpdateTaskImpl getErrorUpdateTask(Exception e) {
     updateTask.setException(e);
     return updateTask;
   }
+
+  private boolean updateIfNewReleaseAvailableIsTaskInProgress() {
+    synchronized (updateIfNewReleaseTaskLock) {
+      return cachedUpdateIfNewReleaseTask != null && !cachedUpdateIfNewReleaseTask.isComplete();
+    }
+  }
+
+  private boolean awaitingSignInDialogConfirmation() {
+    return (updateIfNewReleaseAvailableIsTaskInProgress() && remakeSignInConfirmationDialog);
```

Comment:
Even though it wouldn't make a practical difference, wouldn't it make more sense if this checked if `showSignInDialogTask` was in progress, instead of `cachedUpdateIfNewReleaseTask`?

Reply:
That does make more sense and it shouldn't change anything will change